### ISSUE TITLE
`readr::read_delim()`: delim argument accepts >1 character

### ISSUE THREAD

**Issue by saghirb** [tidyverse/readr#802]
`readr::read_delim()` seems to accept more than one character for the `delim` argument without giving an error or appropriate warning (on the reprex below the warning is for another issue). In the help it states delim is "Single character used to separate fields within a record". Base R `read.delim()` gives an error when `sep`` is not one byte.

```
library(tidyverse)

delimFile <- tempfile(fileext = ".csv")
delimSpace <- c('Date , City , Temp
"18/06/2017" , "Lisbon, Portugal" , 23
"18/06/2017" , "Rome, Italy" , 25
"18/06/2017" , "Oslo, Norway" , 12')
write_file(delimSpace, delimFile)

# I didn't expect this outcome or the warning message 
  space_delim_3 <- readr::read_delim(delimFile, delim = " , ")
  glimpse(space_delim_3)

# Base R read.delim() gives an error when sep is not one byte.
# Not run.
#   space.delim.3 <- read.delim(delimFile, sep = " , ", stringsAsFactors = FALSE)
```

It would be a nice to be able to use more than one character for the `delim` argument to handle CSV files like the one above.

**Comment by cboettig** [comment]
:+1: I'd love to have support for multiple characters as delim, like `vroom` does!  (For example, NCBI dumps use `\t|\t` delimited files).  

Consider:

```r
readr::read_delim("x\t|\ty\n1\t|\t2\n3\t|\t4", "\t|\t")
```

**Comment by jimhester** [comment]
Fixed by #1172
